src/AhRulesBot/AhRulesBot/BotRequestsProcessing/Handlers/CustomRulesMessageHandler.cs

```diff
@@ -42,8 +42,7 @@ private List<string> ProcessRulesRequest(string message)
             Func<CustomRuleItem, bool> messageContains = item => item.Title.Contains(message, StringComparison.InvariantCultureIgnoreCase);
             Func<CustomRuleItem, bool> messageExact = item => item.Title.Equals(message, StringComparison.InvariantCultureIgnoreCase);
 
-            _channel.TryRead(out List<CustomRuleItem> rulesUpdate);
-            if (rulesUpdate != null)
+            while (_channel.TryRead(out List<CustomRuleItem> rulesUpdate) && rulesUpdate != null)
```

Comment:
this would still lead to potential issues with amount of memory used, because it will grow and grow, until somebody eventually hit this handler
but it would not contain bug from core business logic, so approved